docs/source/asr/asr_language_modeling.rst

```diff
@@ -199,4 +199,84 @@ In this approach a neural network is used which can gives scores to a candidate.
 The top K candidates produced by the beam search decoding (beam width of K) are given to a neural language model to rank them.
 Ranking can be done by a language model which gives a score to each candidate.
 This score is usually combined with the scores from the beam search decoding to produce the final scores and rankings.
+
+Train Neural Rescorer
+=====================
+
 An example script to train such a language model with Transformer can be found at `examples/nlp/language_modeling/transformer_lm.py <https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/language_modeling/transformer_lm.py>`__.
+It trains a TransformerLMModel which can be used as a neural rescorer for an ASR system.
+
+
+Evaluation
+==========
+
+Given a trained TransformerLMModel `.nemo` file, this script can be used to re-score beams obtained with ASR model.
+You also need the .tsv file containing the candidates produced by the acoustic model by beam search decoding.
+The candidates could be the result of just the beam search decoding or in fusion with an N-gram LM.
+You may produce this file by specifying `--preds_output_folder' for the `/scripts/asr_language_modeling/ngram_lm/eval_beamsearch_ngram.py`.
+More detail can be found here: `https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/asr/asr_language_modeling.html`.
+The neural rescorer can be used to rescore these candidates with two parameters of alpha and beta as the following:
+
+.. code::
+
+    final_score = beam_search_score + rescorer_alpha*neural_rescorer_score + rescorer_beta*seq_length
+
+Parameter `rescorer_alpha` specifies amount of importance to place on the neural rescorer model, and `rescorer_beta` is
+a penalty term to consider the sequence length in the scores. They have similar effects like the parameters
+`beam_alpha` and `beam_beta` of beam search decoder and N-gram LM.
+
+You may follow the following steps to evaluate a neural LM:
+
+#. Obtain `.tsv` file with beams and their corresponding scores. Scores can be from a regular beam search decoder or
+   in fusion with an N-gram LM scores. For a given beam size `beam_size` and a number of examples
+   for evaluation `num_eval_examples`, it should contain (`beam_size` x `num_eval_examples`) lines of
+   form `beam_candidate_text \t score`. This file can be generated by `scripts/asr_language_modeling/ngram_lm/eval_beamsearch_ngram.py <https://github.com/NVIDIA/NeMo/blob/main/scripts/asr_language_modeling/ngram_lm/eval_beamsearch_ngram.py>`__
```

Comment:
Lines of form? As in formatted lines of samples?

Reply:
Yes, it is a tsv file. Each line should have the following format:
beam_candidate_text \t score

Can I explain it in a better way?

Reply:
A code block with examples would help if possible. This is fine albeit a little confusing at first read.